broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -272,11 +272,17 @@ public void onRecovered() {
         });
   }
 
-  private void onInstallFailure() {
+  private void onInstallFailure(final long term) {
     zeebePartitionHealth.setServicesInstalled(false);
+    context
+        .getPartitionListeners()
+        .forEach(l -> l.onBecomingInactive(context.getPartitionId(), term));
+
     if (context.getRaftPartition().getRole() == Role.LEADER) {
       LOG.info("Unexpected failures occurred when installing leader services, stepping down");
       context.getRaftPartition().stepDown();
+    } else {
+      context.getRaftPartition().leave();
```

Comment:
We should either "stop" the raft server or transition to "Role.INACTIVE". Transitioning to Inactive will be safer, because several other modules might expect a running raft server.

Reply:
That's what `leave()` does:
https://github.com/zeebe-io/zeebe/blob/ba3400c7ee0a6f910a3b1cfef75fb668f7cecca0/atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftClusterContext.java#L262
`leave()` transitions to INACTIVE. The reaction to the role transition is to leave the cluster. Which makes sense since it's not going to take part in consensus. We can shutdown the broker so it can be restarted as soon as possible, but that will impact other partitions

Reply:
`leave()` also leaves the replication group which means that the other nodes would stop communicating with this - we don't want to do that because this node is still part of the replication group, but it is only in a error state.. What we really need is only transition to Inactive which means it is still part of the replication group, but doesn't actively do anything. It acts as if it is unreachable.